base/Base.jl

```diff
@@ -116,7 +116,13 @@ include("range.jl")
 include("error.jl")
 
 # core numeric operations & types
-==(x, y) = x === y
+unwrap_isequal(x) = x
```

Comment:
I think, if we go with this, it should be used much more broadly:
```suggestion
unwrap_for_compare(x) = x
```

Then we need to go update `==`, `<`, `isless`, `isequal`, `hash` etc.

But perhaps instead I should open an issue about eliminating WeakRef (in favor, similar to the `atomic` attribute, of being a runtime-settable attribute in Array flags or in the field properties of a struct). Unfortunately, we cannot delete WeakRef until v2 regardless, since we can give a replacement for it, but we cannot remove it, until then.